src/devices/Mhz19b/Mhz19b.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO.Ports;
+using System.Text;
+using System.Threading;
+using UnitsNet;
+
+namespace Iot.Device.Mhz19b
+{
+    /// <summary>
+    /// Add documentation here
+    /// </summary>
+    public sealed partial class Mhz19b : IDisposable
+    {
+        private enum CommonMessageBytes
+        {
+            Start = 0xff,
+            SensorNumber = 0x01,
+            Empty = 0x00
+        }
+
+        private enum Command : byte
+        {
+            ReadCo2Concentration = 0x86,
+            CalibrateZeroPoint = 0x87,
+            CalibrateSpanPoint = 0x88,
+            AutoCalibrationSwitch = 0x79,
+            DetectionRangeSetting = 0x99
+        }
+
+        private enum MessageFormat
+        {
+            Start = 0x00,
+            SensorNum = 0x01,
+            Command = 0x02,
+            DataHigh = 0x03,
+            DataLow = 0x04,
+            Checksum = 0x08
+        }
+
+        private const int MessageSize = 9;
+
+        private bool _disposed = false;
+        private SerialPort _serialPort = null;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Mhz19b"/> class.
+        /// </summary>
+        /// <param name="uartDevice">Path to the UART device / serial port, e.g. /dev/serial0</param>
+        public Mhz19b(string uartDevice)
+        {
+            if (uartDevice == null)
+            {
+                throw new ArgumentNullException(nameof(uartDevice));
+            }
+
+            // create serial port using the setting acc. to datasheet, pg. 7, sec. general settings
+            _serialPort = new SerialPort(uartDevice, 9600, Parity.None, 8, StopBits.One);
+            _serialPort.Encoding = Encoding.ASCII;
+            _serialPort.ReadTimeout = 1000;
+            _serialPort.WriteTimeout = 1000;
+        }
```

Comment:
Do you have any way to check that the sensor is attached? A serial number to read or equivalent?

Reply:
No, unfortunately there is nothing that can be used to check availability and communication.
If the sensor is not recognizing the request it doesn't respond.